Source/WarpX.cpp

```diff
@@ -1487,13 +1491,40 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     if (deposit_charge)
     {
         rho_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba,rho_nodal_flag),dm,2*ncomps,ngRho,tag("rho_fp"));
+
+        // Also allocate the MultiFab for the full phi grid
+        // get the full domain
+        Box domain_box = Geom(lev).Domain();
+        // make a new BoxArray from the domain Box
+        BoxArray ba_full(domain_box);
+        // make a DistributionMapping from the new BoxArray
+        DistributionMapping dm_full;
+        // force that distribution mapping to only go to the root proc
+        Vector<int> pmap = {0};
+        dm_full.define(pmap);
+        // make a FabArray to hold the phi data
+        full_rho_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba_full,rho_nodal_flag),dm_full,2*ncomps,ngRho,tag("full_rho_fp"));
     }
 
     if (do_electrostatic == ElectrostaticSolverAlgo::LabFrame)
     {
         IntVect ngPhi = IntVect( AMREX_D_DECL(1,1,1) );
         phi_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba,phi_nodal_flag),dm,ncomps,ngPhi,tag("phi_fp"));
         phi_fp[lev]->setVal(0.);
+
+        // Also allocate the MultiFab for the full phi grid
+        // get the full domain
+        Box domain_box = Geom(lev).Domain();
+        // make a new BoxArray from the domain Box
+        BoxArray ba_full(domain_box);
+        // make a DistributionMapping from the new BoxArray
+        DistributionMapping dm_full;
+        // force that distribution mapping to only go to the root proc
+        Vector<int> pmap = {0};
+        dm_full.define(pmap);
+        // make a FabArray to hold the phi data
+        full_phi_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba_full,phi_nodal_flag),dm_full,ncomps,ngPhi,tag("full_phi_fp"));
```

Comment:
Just curious why does `full_rho_fp` use a MultiFab with `2*ncomps` as a parameter while `full_phi_fp` uses one with just `ncomps`?

Reply:
I have no idea. We just followed what was done in `rho_fp`, which also uses `2*ncomps`, but I don't know why.